File: backend/app/api/assistant.py
from fastapi import APIRouter
from pydantic import BaseModel

router = APIRouter(prefix="/v1", tags=["assistant"])


class AssistantRequest(BaseModel):
    engagement_id: str
    message: str


@router.post("/assistant/chat")
def assistant_chat(payload: AssistantRequest) -> dict[str, str]:
    return {
        "engagement_id": payload.engagement_id,
        "answer": "Assistant integration is a placeholder in this skeleton."
    }

File: frontend/lib/api/assistant.ts
import { apiRequest } from "./client";

export interface AssistantResponse {
  engagement_id: string;
  answer: string;
}

export function sendAssistantMessage(
  engagementId: string,
  message: string
): Promise<AssistantResponse> {
  return apiRequest<AssistantResponse>("/v1/assistant/chat", {
    method: "POST",
    body: JSON.stringify({ engagement_id: engagementId, message })
  });
}
